src/stdatamodels/model_base.py

```diff
@@ -113,6 +114,13 @@ def __init__(self, init=None, schema=None, memmap=False,
             If `None`, value will be taken from the environmental STRICT_VALIDATION.
             Otherwise, the default value is `False`.
 
+        validate_on_assignment : bool
+            Defaults to 'True'.
+            If 'True', attribute assignments are validated at the time of assignment.
+            Validation errors generate warnings.
+            If 'False', schema validation occurs only once at the time of write.
+            Validation errors generate warnings.
```

Comment:
You'll want to update this to `bool or None` and explain what the default of None means.  Basically, it will follow the same pattern as the `strict_validation` kwarg.

Reply:
I expanded the docstring to explain the usage of and default to `None`.